cmd/katib-controller/v1beta1/main.go

```diff
@@ -56,6 +58,9 @@ func main() {
 	flag.BoolVar(&enableGRPCProbeInSuggestion, "enable-grpc-probe-in-suggestion", true, "enable grpc probe in suggestions")
 	flag.Var(&trialResources, "trial-resources", "The list of resources that can be used as trial template, in the form: Kind.version.group (e.g. TFJob.v1.kubeflow.org)")
 	flag.IntVar(&webhookPort, "webhook-port", 8443, "The port number to be used for admission webhook server.")
+	// For leader election
+	flag.BoolVar(&enableLeaderElection, "leader-elect", true, "Enable leader election for katib-controller. Enabling this will ensure there is only one active katib-controller.")
```

Comment:
Should we use leader-election here?
```suggestion
	flag.BoolVar(&enableLeaderElection, "leader-election", true, "Enable leader election for katib-controller. Enabling this will ensure there is only one active katib-controller.")
```

Reply:
I created a demo project using`kubebuilder init`, and the generated code looks like this:

```
func main() {
	var metricsAddr string
	var enableLeaderElection bool
	flag.StringVar(&metricsAddr, "metrics-addr", ":8080", "The address the metric endpoint binds to.")
	flag.BoolVar(&enableLeaderElection, "enable-leader-election", false,
		"Enable leader election for controller manager. "+
			"Enabling this will ensure there is only one active controller manager.")
	flag.Parse()

	ctrl.SetLogger(zap.New(zap.UseDevMode(true)))

	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
		Scheme:             scheme,
		MetricsBindAddress: metricsAddr,
		Port:               9443,
		LeaderElection:     enableLeaderElection,
		LeaderElectionID:   "956b9520.example.org",
	})
	if err != nil {
		setupLog.Error(err, "unable to start manager")
		os.Exit(1)
	}

	// +kubebuilder:scaffold:builder

	setupLog.Info("starting manager")
	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
		setupLog.Error(err, "problem running manager")
		os.Exit(1)
	}
}
```

Reply:
Maybe enable-leader-election is better. WDYT

We keep consistent with default name generated by kubebuilder.

Reply:
Done.